clients/src/main/java/org/apache/kafka/common/requests/MetadataResponse.java

```diff
@@ -91,7 +91,7 @@ public int throttleTimeMs() {
     public Map<String, Errors> errors() {
         Map<String, Errors> errors = new HashMap<>();
         for (MetadataResponseTopic metadata : data.topics()) {
-            if (metadata.errorCode() != Errors.NONE.code())
+            if (metadata.errorCode() != Errors.NONE.code() && metadata.name() != null)
```

Comment:
What happens if `name` is null - won't we quietly ignore? Would it be better to throw an exception since we don't want anyone using this with topic ids?

Reply:
Firstly I want to leave those check to callers, now I think it's better to throw an exception, So I add another method errorsByTopicId.